From 8556933f52df71cf1b58eeaeae1865ebf7c005cc Mon Sep 17 00:00:00 2001
From: "Thomas E. Dickey" <dickey@invisible-island.net>
Date: Sun, 3 Jun 2018 01:08:44 +0000
Subject: [PATCH] ncurses 6.1 - patch 20180602

+ add RPM test-package "ncursest-examples".
+ modified RPM test-package to work with Mageia6.
---
 NEWS                               |  7 ++--
 VERSION                            |  2 +-
 dist.mk                            |  4 +--
 package/debian-mingw/changelog     |  4 +--
 package/debian-mingw64/changelog   |  4 +--
 package/debian/changelog           |  4 +--
 package/mingw-ncurses.nsi          |  4 +--
 package/mingw-ncurses.spec         |  2 +-
 package/ncurses.spec               | 19 +++++++++-
 test/package/ncurses-examples.spec | 57 +++++++++++++++++++++++++++---
 10 files changed, 87 insertions(+), 20 deletions(-)

diff --git a/NEWS b/NEWS
index 1df56e55..39942d12 100644
--- a/NEWS
+++ b/NEWS
@@ -25,7 +25,7 @@
 -- sale, use or other dealings in this Software without prior written        --
 -- authorization.                                                            --
 -------------------------------------------------------------------------------
--- $Id: NEWS,v 1.3134 2018/05/26 20:27:35 tom Exp $
+-- $Id: NEWS,v 1.3137 2018/06/02 22:52:28 tom Exp $
 -------------------------------------------------------------------------------
 
 This is a log of changes that ncurses has gone through since Zeyd started
@@ -45,6 +45,10 @@ See the AUTHORS file for the corresponding full names.
 Changes through 1.9.9e did not credit all contributions;
 it is not possible to add this information.
 
+20180602
+	+ add RPM test-package "ncursest-examples".
+	+ modified RPM test-package to work with Mageia6.
+
 20180526
 	+ add note in curs_util.3x about unctrl.h
 	+ review/improve header files to ensure that those include necessary
@@ -4126,7 +4130,6 @@ it is not possible to add this information.
 	+ improved configure macros CF_GCC_ATTRIBUTES, CF_PROG_LINT.
 
 20091114
-
 	+ updated man/curs_trace.3x
 	+ limit hashing for termcap-names to 2-characters (Ubuntu #481740).
 	+ change a variable name in lib_newwin.c to make it clearer which
diff --git a/VERSION b/VERSION
index b5635be0..c075ebaa 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-5:0:10	6.1	20180526
+5:0:10	6.1	20180602
diff --git a/dist.mk b/dist.mk
index 4cfe5226..089fadab 100644
--- a/dist.mk
+++ b/dist.mk
@@ -25,7 +25,7 @@
 # use or other dealings in this Software without prior written               #
 # authorization.                                                             #
 ##############################################################################
-# $Id: dist.mk,v 1.1225 2018/05/20 18:12:00 tom Exp $
+# $Id: dist.mk,v 1.1226 2018/05/28 19:47:08 tom Exp $
 # Makefile for creating ncurses distributions.
 #
 # This only needs to be used directly as a makefile by developers, but
@@ -37,7 +37,7 @@ SHELL = /bin/sh
 # These define the major/minor/patch versions of ncurses.
 NCURSES_MAJOR = 6
 NCURSES_MINOR = 1
-NCURSES_PATCH = 20180526
+NCURSES_PATCH = 20180602
 
 # We don't append the patch to the version, since this only applies to releases
 VERSION = $(NCURSES_MAJOR).$(NCURSES_MINOR)
diff --git a/package/debian-mingw/changelog b/package/debian-mingw/changelog
index 7162c2ea..02dc321e 100644
--- a/package/debian-mingw/changelog
+++ b/package/debian-mingw/changelog
@@ -1,8 +1,8 @@
-ncurses6 (6.1+20180526) unstable; urgency=low
+ncurses6 (6.1+20180602) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sun, 20 May 2018 14:12:00 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Mon, 28 May 2018 15:47:08 -0400
 
 ncurses6 (5.9-20131005) unstable; urgency=low
 
diff --git a/package/debian-mingw64/changelog b/package/debian-mingw64/changelog
index 7162c2ea..02dc321e 100644
--- a/package/debian-mingw64/changelog
+++ b/package/debian-mingw64/changelog
@@ -1,8 +1,8 @@
-ncurses6 (6.1+20180526) unstable; urgency=low
+ncurses6 (6.1+20180602) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sun, 20 May 2018 14:12:00 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Mon, 28 May 2018 15:47:08 -0400
 
 ncurses6 (5.9-20131005) unstable; urgency=low
 
diff --git a/package/debian/changelog b/package/debian/changelog
index 47b7c9cf..fda2ce04 100644
--- a/package/debian/changelog
+++ b/package/debian/changelog
@@ -1,8 +1,8 @@
-ncurses6 (6.1+20180526) unstable; urgency=low
+ncurses6 (6.1+20180602) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sun, 20 May 2018 14:12:00 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Mon, 28 May 2018 15:47:08 -0400
 
 ncurses6 (5.9-20120608) unstable; urgency=low
 
diff --git a/package/mingw-ncurses.nsi b/package/mingw-ncurses.nsi
index 9e607c27..f7e6a0bb 100644
--- a/package/mingw-ncurses.nsi
+++ b/package/mingw-ncurses.nsi
@@ -1,4 +1,4 @@
-; $Id: mingw-ncurses.nsi,v 1.272 2018/05/20 18:12:00 tom Exp $
+; $Id: mingw-ncurses.nsi,v 1.273 2018/05/28 19:47:08 tom Exp $
 
 ; TODO add examples
 ; TODO bump ABI to 6
@@ -10,7 +10,7 @@
 !define VERSION_MAJOR "6"
 !define VERSION_MINOR "1"
 !define VERSION_YYYY  "2018"
-!define VERSION_MMDD  "0526"
+!define VERSION_MMDD  "0602"
 !define VERSION_PATCH ${VERSION_YYYY}${VERSION_MMDD}
 
 !define MY_ABI   "5"
diff --git a/package/mingw-ncurses.spec b/package/mingw-ncurses.spec
index d5aa075d..336e81f0 100644
--- a/package/mingw-ncurses.spec
+++ b/package/mingw-ncurses.spec
@@ -3,7 +3,7 @@
 Summary: shared libraries for terminal handling
 Name: mingw32-ncurses6
 Version: 6.1
-Release: 20180526
+Release: 20180602
 License: X11
 Group: Development/Libraries
 Source: ncurses-%{version}-%{release}.tgz
diff --git a/package/ncurses.spec b/package/ncurses.spec
index 6b99d784..61f7dee9 100644
--- a/package/ncurses.spec
+++ b/package/ncurses.spec
@@ -1,7 +1,7 @@
 Summary: shared libraries for terminal handling
 Name: ncurses6
 Version: 6.1
-Release: 20180526
+Release: 20180602
 License: X11
 Group: Development/Libraries
 Source: ncurses-%{version}-%{release}.tgz
@@ -38,7 +38,21 @@ This package is used for testing ABI %{MY_ABI} with POSIX threads.
 
 %prep
 
+%global is_mandriva %(test -f /etc/mandriva-release && echo 1 || echo 0)
+%global is_redhat   %(test -f /etc/redhat-release && echo 1 || echo 0)
+%global is_suse     %(test -f /etc/SuSE-release && echo 1 || echo 0)
+
+# nor are debug-symbols
 %define debug_package %{nil}
+
+%if %{is_mandriva}
+%define _disable_ld_as_needed 1
+%define _disable_ld_no_undefined 1
+# libtool is not used here...
+%define _disable_libtoolize 1
+%define _disable_ld_build_id 1
+%endif
+
 %setup -q -n ncurses-%{version}-%{release}
 
 %build
@@ -161,6 +175,9 @@ rm -rf $RPM_BUILD_ROOT
 
 %changelog
 
+* Sat Jun 02 2018 Thomas E. Dickey
+- build-fix for Mageia
+
 * Sat May 26 2018 Thomas E. Dickey
 - use predefined configure-macro
 - separate ncurses6/ncursest6 packages
diff --git a/test/package/ncurses-examples.spec b/test/package/ncurses-examples.spec
index 8ca1c1e9..a3958cde 100644
--- a/test/package/ncurses-examples.spec
+++ b/test/package/ncurses-examples.spec
@@ -1,8 +1,9 @@
-Summary: ncurses-examples - example/test programs from ncurses
+Summary: example/test programs from ncurses
 %define AppProgram ncurses-examples
+%define AltProgram ncursest-examples
 %define AppVersion MAJOR.MINOR
 %define AppRelease YYYYMMDD
-# $Id: ncurses-examples.spec,v 1.11 2018/01/15 16:14:16 tom Exp $
+# $Id: ncurses-examples.spec,v 1.12 2018/06/02 22:46:44 tom Exp $
 Name: %{AppProgram}
 Version: %{AppVersion}
 Release: %{AppRelease}
@@ -16,8 +17,20 @@ Packager: Thomas Dickey <dickey@invisible-island.net>
 These are the example/test programs from the ncurses MAJOR.MINOR distribution,
 for patch-date YYYYMMDD.
 
-This package installs in "bin/ncurses-examples" to avoid conflict with other
+This package installs in "bin/%{AppProgram}" to avoid conflict with other
 packages.
+
+%package -n %{AltProgram}
+Summary:  examples/test programs from ncurses with POSIX thread support
+
+%description -n %{AltProgram}
+These are the example/test programs from the ncurses MAJOR.MINOR distribution,
+for patch-date YYYYMMDD, using the "ncurseswt" library to demonstrate the
+use of POSIX threads, e.g., in ditto, rain, and worm.
+
+This package installs in "bin/%{AltProgram}" to avoid conflict with other
+packages.
+
 %prep
 
 %setup -q -n %{AppProgram}-%{AppRelease}
@@ -26,8 +39,14 @@ packages.
 
 %build
 
+%global _configure ../configure
+%define my_srcdir ..
+
+mkdir BUILD-%{AppProgram}
+pushd BUILD-%{AppProgram}
 INSTALL_PROGRAM='${INSTALL}' \
 NCURSES_CONFIG_SUFFIX=dev \
+CONFIGURE_TOP=%{my_srcdir} \
 %configure \
 	--target %{_target_platform} \
 	--prefix=%{_prefix} \
@@ -37,20 +56,48 @@ NCURSES_CONFIG_SUFFIX=dev \
 	--disable-rpath-hack
 
 make
+popd
+
+mkdir BUILD-%{AltProgram}
+pushd BUILD-%{AltProgram}
+INSTALL_PROGRAM='${INSTALL}' \
+NCURSES_CONFIG_SUFFIX=dev \
+CONFIGURE_TOP=%{my_srcdir} \
+%configure \
+	--target %{_target_platform} \
+	--prefix=%{_prefix} \
+	--bindir=%{_bindir}/%{AltProgram} \
+	--datadir=%{_datadir}/%{AltProgram} \
+	--with-screen=ncursestw6 \
+	--disable-rpath-hack
+
+make
+popd
 
 %install
 [ "$RPM_BUILD_ROOT" != "/" ] && rm -rf $RPM_BUILD_ROOT
 
-make install               DESTDIR=$RPM_BUILD_ROOT
+pushd BUILD-%{AppProgram}
+make install DESTDIR=$RPM_BUILD_ROOT
+popd
+
+pushd BUILD-%{AltProgram}
+make install DESTDIR=$RPM_BUILD_ROOT
+popd
 
 %clean
 [ "$RPM_BUILD_ROOT" != "/" ] && rm -rf $RPM_BUILD_ROOT
 
-%files
+%files -n %{AppProgram}
 %defattr(-,root,root)
 %{_bindir}/%{AppProgram}/*
 %{_datadir}/%{AppProgram}/*
 
+%files -n %{AltProgram}
+%defattr(-,root,root)
+%{_bindir}/%{AltProgram}/*
+%{_datadir}/%{AltProgram}/*
+
 %changelog
 # each patch should add its ChangeLog entries here
 
